### PR TITLE
Add extra_headers argument to LivyHook and LivyOperator

### PR DESCRIPTION
**Pull request by pgagnon**
Adds an `extra_headers` option to `LivyHook` and `LivyOperator`. This allows passing extra header options to requests before sending a request to Livy. This is required for instance when CSRF protection is enabled (i.e. on HDinsights): https://github.com/MicrosoftDocs/azure-docs/issues/10457.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: tests/providers/apache/livy/hooks/test_livy.py [hunk 1]
```diff
@@ -316,6 +316,20 @@ def test_invalid_uri(self):
...
+            'POST',
+            '//livy:8998/batches',
+            json={'id': BATCH_ID, 'state': BatchState.STARTING.value, 'log': []},
+            status_code=201,
+            request_headers={'X-Requested-By': 'user'}
+        )
+
+        hook = LivyHook(extra_headers={'X-Requested-By': 'user'})
+        hook.post_batch(file='sparkapp')
+
```
Comment (kaxil) [inline tests/providers/apache/livy/hooks/test_livy.py]: What does this test check?

Comment (pgagnon) [reply tests/providers/apache/livy/hooks/test_livy.py]: It checks that `X-Requested-By: user` is present in the request headers.

edit: ref: https://requests-mock.readthedocs.io/en/latest/matching.html#request-headers

Comment (kaxil) [reply tests/providers/apache/livy/hooks/test_livy.py]: Tested it locally, works fine

**Review by kaxil — APPROVED** [review]

**Comment by kaxil** [comment]
(re-triggering CI)

**Comment by kaxil** [comment]
@pgagnon this PR is 461 commits behind the `main` branch

**Comment by kaxil** [comment]
I have just rebased & force-pushed !!

**Comment by pgagnon** [comment]
> @pgagnon this PR is 461 commits behind the `main` branch

Sorry I didn't realize I was so far behind 😭😂

**Comment by kaxil** [comment]
Some static checks are failing: https://github.com/apache/airflow/pull/16512/checks?check_run_id=2853948767#step:10:265

Would suggest installing pre-commits hook to run this automatically locally:

https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst

**Comment by pgagnon** [comment]
@kaxil Ran the hooks, rebased and force-pushed.

File: tests/providers/apache/livy/hooks/test_livy.py [hunk 2]
```diff
@@ -255,7 +255,7 @@ def test_post_batch_arguments(self, mock_request):
         resp = hook.post_batch(file='sparkapp')
 
         mock_request.assert_called_once_with(
-            method='POST', endpoint='/batches', data=json.dumps({'file': 'sparkapp'})
+            method='POST', endpoint='/batches', data=json.dumps({'file': 'sparkapp'}), headers={}
```
Comment (pgagnon) [inline tests/providers/apache/livy/hooks/test_livy.py:258]: @kaxil I modified this test slightly. Just wanted to explicitly point it out.

**Comment by pgagnon** [comment]
Retriggering CI

**Review by kaxil — APPROVED** [review]

**Pull request merged**